src/main/java/com/rybka/service/ApiConnectorService.java

```diff
@@ -0,0 +1,37 @@
+package com.rybka.service;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.rybka.exception.CurrencyAPICallException;
+import com.rybka.model.ExchangeResponse;
+
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+
+public class ApiConnectorService {
+
+    private static final String API_KEY = "4e0bbdc44fcdf05612fa0882";
+    private final HttpClient client = HttpClient.newHttpClient();
+    private final ObjectMapper mapper = new ObjectMapper();
+
+    public ExchangeResponse retrieveRates(String userBaseCurrency) {
+        String url = String.format("https://prime.exchangerate-api.com/v5/%s/latest/%s",
+                API_KEY, userBaseCurrency);
+        try {
+            HttpRequest request = HttpRequest.newBuilder()
+                    .uri(URI.create(url))
+                    .build();
+            var response = client.send(request,
+                    HttpResponse.BodyHandlers.ofString());
+
+            JsonNode tree = mapper.readTree(response.body());
+
+            return mapper.treeToValue(tree, ExchangeResponse.class);
```

Comment:
mapper.readValues(response.body(), ExchangeResponse.class) doesn't work here?

Reply:
As far as I remember it works with json as String for parameters, so you'll be able to get rid of JsonNode tree.

Reply:
fixed